src/Sylius/Component/Core/Promotion/Checker/TaxonPromotionRuleChecker.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Promotion\Checker;
+
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Core\Promotion\Checker\Rule\TotalOfItemsFromTaxonRuleChecker;
+use Sylius\Component\Promotion\Model\PromotionRuleInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+final class TaxonPromotionRuleChecker implements TaxonPromotionRuleCheckerInterface
```

Comment:
```suggestion
final class TaxonInPromotionUsageChecker implements TaxonInPromotionUsageCheckerInterface
```
?

In my opinion, this class name better reflects its purpose.